libvncserver/ws_decode.h

```diff
@@ -0,0 +1,151 @@
+#ifndef _WS_DECODE_H_
```

Comment:
Not overly complex, but also not a complete websocket implementation (afaik, it's was and still is a counterpart of the websocket-subset noVNC uses/used at the client side). And as we have seen, it does offer enough potential for bugs and/or security-relevant overflow issues.